### ISSUE TITLE
reproduce the training

### ISSUE THREAD

**Issue by pengzhangzhi** [andrew-cr/tauLDR#4]
Hi @andrew-cr 

I tried to reproduce the training on cifar-1o and obtained the following losses.

It seems to me a little weird that the neg_elbo term is sooo large compared to the NLL terms. Am I wrong? FYI, I changed the batch size to 40 to save memory consumption.  Do you guys also obtain similar values in the initial training stage?
```
NLL: 8.108, neg_elbo: 12571671.000
```

Best,
Zhangzhi

**Comment by andrew-cr** [comment]
Hi @pengzhangzhi 

Thanks for your interest in our work! You are right that the neg_elbo term is quite large and we also see this. We removed many constant terms that don't depend on the model parameters when writing down the optimization objective and so the absolute value of the objective that is left can be quite large. We didn't experiment much with changing the scalings of the various terms in the optimization objective as we found the first setting tried had reasonable performance and did not perform further sweeps. But it could be the case that performance could be improved if the loss scalings are tuned. 

Please let us know if you have any further issues with running the training :) 

Andrew

**Comment by pengzhangzhi** [comment]
Thanks! 1. Would you explain intuitively what the term neg_elbo does?  2. Do you think removing it would hurt the model performance?

**Comment by andrew-cr** [comment]
So neg_elbo is a bound on the model negative log-likelihood assigned to the datapoints under the reverse generative process defined by the denoising model. So minimizing the negative log likelihood will maximize the model log-likelihood assigned to the datapoints thus making the model better. The direct NLL term is a direct cross entropy loss on the denoising model output which we show in our paper https://arxiv.org/pdf/2205.14987v2.pdf Appendix D Proposition 9 is actually a bound on the negative elbo in the discrete time case. The two objectives neg_elbo and direct NLL have the same optimum and so both work reasonably well but they will likely have different dynamics due to penalizing model errors differently. We train with the combined loss of neg_elbo and NLL as it was found in https://arxiv.org/pdf/2107.03006.pdf Section 6 that training with purely the direct NLL achieved worse performance in the long run.

**Comment by pengzhangzhi** [comment]
Thanks for the explanation! Very helpful :)
